### PR TITLE
Cloud Function for Vision API

### PR DESCRIPTION
**Pull request by shubham-ture**
# Description

Google cloud function for running Vision API on image file.

# Fixes # (issue)

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

-   [X] I have tested my code locally
-   [X] I have tested my code in staging server

# Checklist

-   [X] My Pull request is having proper title
-   [X] My Commits are organised
-   [X] My Pull is not having any merge commits
-   [X] I have performed a self-review of my own code
-   [X] I have evaluated the impact of the change
-   [X] I have followed the syntax standards
-   [X] My code is properly separated between different layers (Controller, Business Logic & Repository)

### CODE REVIEW

File: google_vision_python/constants.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+google_vision_api_key_path = 'your_vision_api_key.json'
+pubsub_key = 'your_pubsub_api_key.json'
+
+success_response = 'success'
+failure_response = 'failed'
+
+success_status_code = 200
+internal_server_error_code = 500
+
+pubsub_project_id = 'oceanic-bindery-327007'
```
Comment (sagar-vaghela97) [inline google_vision_python/constants.py:10]:
```suggestion
pubsub_project_id = 'your_project_id'
```

File: google_vision_python/main.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    bucket_credentials = google.oauth2.credentials.Credentials(bucket_access_token)
+
+    source_file_path = str(source_path) + '/' + str(file_name)
+
+    logging.error("Downloading file from bucket")
+    temp_file_path = download_from_bucket(bucket_name, file_name, project_id, bucket_credentials, source_file_path)
+
+    time.sleep(2)
+
```
Comment (sagar-vaghela97) [inline google_vision_python/main.py:59]: File will be downloaded in 2 sec?

Comment (sagar-vaghela97) [reply google_vision_python/main.py:59]: Can we go to the next step once the file is downloaded?

Comment (shubham-ture) [reply google_vision_python/main.py:59]: > Can we go to the next step once the file is downloaded?

?

Comment (shubham-ture) [reply google_vision_python/main.py:59]: > File will be downloaded in 2 sec?

Yes, for the most part 2 seconds is enough.

Comment (sagar-vaghela97) [reply google_vision_python/main.py:59]: We cannot assume all sizes of files will be downloaded in 2 seconds. Kindly make it async

Comment (shubham-ture) [reply google_vision_python/main.py:59]: by sync do you mean to remove the sleep timer?

Comment (sagar-vaghela97) [reply google_vision_python/main.py:59]: Below code should run only after the file is downloaded..

  with io.open(temp_file_path, 'rb') as image:
        image_data = image.read()
    vision_api_json_response = run_vision_api_on_image(image_data)

Comment (sagar-vaghela97) [reply google_vision_python/main.py:59]: https://googleapis.dev/python/storage/latest/blobs.html

File: google_vision_python/main.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+def delete_files_from_directory(directory):
+    for f in os.listdir(directory):
+        os.remove(os.path.join(directory, f))
+
+
+def download_from_bucket(bucket_name, file_name, project_id, bucket_credentials, source_file_path):
+
+    #os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = constants.bucket_key
+    #storage_client = storage.Client()
```
Comment (sagar-vaghela97) [inline google_vision_python/main.py:139]: Remove commneted codes

File: google_vision_python/main.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        "fileName" : file_name_json_format,
+        "status": status
+    }
+    return response_object
+
+
+def publish_to_pubsub_topic(project_id, topic_name, data):
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = constants.pubsub_key
+    publisher = pubsub_v1.PublisherClient()
+    logging.error(f'Publishing message to topic {topic_name}')
```
Comment (sagar-vaghela97) [inline google_vision_python/main.py:108]: It should be logging.info or use this

https://stackoverflow.com/questions/54147483/google-cloud-functions-python-logging-issue

**Review by sagar-vaghela97 — APPROVED** [review]

**Pull request merged**